### PR TITLE
Update `fly sync` to only download on version mismatch

### LINKED ISSUE #87: Make fly sync faster when no changes
We have some tooling that uses `fly` which calls `fly sync` to make sure that we have the correct version of the client. However, downloading the whole binary each time is quite slow, so we're skipping it if the binary is <24hrs old.

It looks like `fly sync` could be sped up by sending an `If-Modified-Since` header with the current binary's mtime. `client.GetCLIReader` uses `http.ServeFile` which already does the comparison and returns 304 if the server's file has an earlier mtime.

Does this sound like a good idea?

I haven't made a PR because I'm not clear how you'd prefer it to be implemented. Should `client.GetCLIReader` take a time argument and return an empty body if the response code is 304? Should there be an integration test between ATC and fly, using the real handler?

### PR DESCRIPTION
**Pull request by geofffranks**
When `fly sync` is called, it will only download the
new executable, if the `fly` binary is not the same
version as  the ATC version.

Fixes #87

### CODE REVIEW

**Comment by lnguyen** [comment]
:+1:

File: commands/sync.go [hunk 1]
```diff
@@ -20,6 +21,15 @@ func (command *SyncCommand) Execute(args []string) error {
 	if err != nil {
 		return err
 	}
+	info, err := target.Client().GetInfo()
+	if err != nil {
+		return err
+	}
+
+	if info.Version == version.Version {
+		fmt.Printf("Good news! You already have the appropriate fly binary for this target.\n")
```
Comment (vito) [inline commands/sync.go]: minor UX thing: this should probably just say `version already matches; skipping`

we tend to stick with terse lowercase text elsewhere

**Comment by vito** [comment]
Looks good; left one inline comment for a wording change. Thanks!

**Comment by geofffranks** [comment]
Wording changed. Thanks!

**Pull request merged**